### PR TITLE
CP-1093 Increase HTTP coverage. Consistent test naming.

### PR DESCRIPTION
**Pull request by evanweible-wf**
## Issue
- Still a few gaps in test coverage around HTTP code
- Integration tests have a pattern for test names but the rest of the tests don't follow this

## Changes
**Source:**
- Fixed an issue where types were being checked in the wrong order (`Blob` vs `File`)
- Browser requests (XHR) now set `responseType = 'blob'` for responses that should be streamed and a `FileReader` is used to obtain the response in bytes. This prevents the browser from having to decode the response which previously was immediately being encoded back into bytes for this use case.

**Tests:**
- Additional tests around HTTP code
- Made test naming consistent for all tests, now includes:
  - test type (unit, integration)
  - topic (HTTP, WebSocket, Mocks, PlatformAdapter)
  - platform (Browser, VM, Mock - integration tests only).

## Areas of Regression
- File Upload (covered by file transfer example)
- Streamed responses (covered by tests)

## Testing
- CI passes
- File transfer example works as expected

## Code Review
@trentgrover-wf
@maxwellpeterson-wf 
@dustinlessard-wf
@jayudey-wf

### CODE REVIEW

**Comment by evanweible-wf** [comment]
The diffs look worse than they are.. I recommend viewing them without whitespace changes: https://github.com/Workiva/w_transport/pull/72/files?w=0

**Comment by dustinlessard-wf** [comment]
Only 67 files changed... damn, was hoping for a high-score. :-)

**Comment by evanweible-wf** [comment]
only about 100 files short of #67  :smile:

**Comment by maxwellpeterson-wf** [comment]
+1 nice!

File: test/naming.dart [hunk 1]
```diff
@@ -19,19 +19,27 @@ const String platformBrowser = 'browser';
...
+const String topicMocks = 'Mocks';
+const String topicPlatformAdapter = 'Platform Adapter';
+const String topicWebSocket = 'WS';
+
+class Naming {
+  String platform;
+  String testType;
+  String topic;
+
+  @override
```
Comment (dustinlessard-wf) [inline test/naming.dart]: I was curious if one must include `@override` with the toString function.  It turns out that it's not necessary : https://dartpad.dartlang.org/053c01d956e344df2f67
No biggie including it though.

**Comment by dustinlessard-wf** [comment]
+1, I found myself quite a few times thinking : "Oh cool, I like that - I think I'm going to copy that approach"

File: test/naming.dart (hunk 1 above)
Comment (evanweible-wf) [inline test/naming.dart]: It's actually never necessary, the benefit you get from it is if you include it on something that doesn't actually override something from a parent class. So if you have a typo or if you refactor something in a parent class and forget to change the child class, the compiler can warn you.

Comment (dustinlessard-wf) [inline test/naming.dart]: Thanks - that makes sense.  I had a feeling you had a reason for including it!

File: lib/src/http/browser/request_mixin.dart [hunk 2]
```diff
@@ -81,9 +80,10 @@ abstract class BrowserRequestMixin implements BaseRequest, CommonRequest {
         c.complete(_createResponse(streamResponse: streamResponse));
       }
     });
-    void onError(error) {
+    onError(error) async {
```
Comment (trentgrover-wf) [inline lib/src/http/browser/request_mixin.dart]: might as well toss a return type on this

**Comment by trentgrover-wf** [comment]
+1

File: lib/src/http/browser/request_mixin.dart (hunk 2 above)
Comment (evanweible-wf) [inline lib/src/http/browser/request_mixin.dart]: added

**Comment by trentgrover-wf** [comment]
+1

**Comment by maxwellpeterson-wf** [comment]
+1

**Comment by evanweible-wf** [comment]
@jayudey-wf ready for merge

**Comment by jayudey-wf** [comment]
QA Resource Approval: +10

- [ ] Testing instruction
- [ ] Dev +1's
- [ ] Dev/QA +10 with detail of what was tested
 - tests pass, file download example works as expected
- [ ] Unit test created/updated
- [ ] All unit tests pass

Merging into master.

**Pull request merged**